static/app/views/settings/account/apiNewToken.tsx

```diff
@@ -19,115 +20,116 @@ import TextBlock from 'sentry/views/settings/components/text/textBlock';
 import PermissionSelection from 'sentry/views/settings/organizationDeveloperSettings/permissionSelection';
 
 const API_INDEX_ROUTE = '/settings/account/api/auth-tokens/';
-type State = {
-  name: string | null;
-  newToken: NewInternalAppApiToken | null;
-  permissions: Permissions;
-};
 
-export default class ApiNewToken extends Component<{}, State> {
-  constructor(props: {}) {
-    super(props);
-    this.state = {
-      name: null,
-      permissions: {
-        Event: 'no-access',
-        Team: 'no-access',
-        Member: 'no-access',
-        Project: 'no-access',
-        Release: 'no-access',
-        Organization: 'no-access',
-        Alerts: 'no-access',
-      },
-      newToken: null,
-    };
-  }
+export default function ApiNewToken() {
+  const [permissions, setPermissions] = useState<Permissions>({
+    Event: 'no-access',
+    Team: 'no-access',
+    Member: 'no-access',
+    Project: 'no-access',
+    Release: 'no-access',
+    Organization: 'no-access',
+    Alerts: 'no-access',
+  });
+  const [newToken, setNewToken] = useState<NewInternalAppApiToken | null>(null);
+  const [preview, setPreview] = useState<string>('');
 
-  onCancel = () => {
-    browserHistory.push(normalizeUrl(API_INDEX_ROUTE));
+  const getPreview = () => {
+    let previewString = '';
+    for (const k in permissions) {
+      if (permissions[k] !== 'no-access') {
+        previewString += `${k.toLowerCase()}:${permissions[k]}\n`;
+      }
+    }
+    return previewString;
   };
```

Comment:
We can remove this state i think... here's a couple moves i'd make:
- Remove `const [preview, setPreview] = useState<string>('');` we don't need that state.
- Move `const getPreview = () => { ... }` up out of the component
  - Give it an argument: `(permissions: Permissions) => { ...`
- Down below we don't need `setPreview(getPreview());` inside `<PermissionSelection>` anymore
- Instead, `<TextareaField>` can have the prop `value={getPreview(permissions)}`

Reply:
i tried this but it doesn't work unfortunately. `value` also doesn't populate the text field as expected, i have to use `state` with `placeholder` in order for anything to display :(

Reply:
this is surprising, but our forms are all too much magic :(